## Практическое задание N2

### Урок 2. Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона

In [6]:
from math import factorial

In [16]:
import numpy as np


def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))


def bi_rasp(n, k, p):
    return combinations(n, k) * (p ** k) * ((1 - p) ** (n - k))


def puasson(m, l):
    return (l ** m) / factorial(m) * np.e ** (-l)



1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.


In [8]:
C85_100 = combinations(100, 85)
C85_100

253338471349988640

In [9]:
Bp = C85_100 * (0.8) ** 85 * (0.2) ** 15
Bp

0.048061793700746556

In [10]:
Bp = bi_rasp(100, 85, 0.8)
Bp

0.048061793700746355

> Вероятность попадания равна 0.048

2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

> a) Так как вероятность наступления события маленькая (0.0004), а событий много (5000) можно применить распределение Пуассона. Считаем от противного - наступление события перегорела хотябы одна лампочка.
>

In [19]:
m = 1
lamda = 1 / 5000
Pm = puasson(m, lamda)
Pm

0.00019996000399973335

In [14]:
1 - Pm

0.9998000399960003

> Вероятность того что лампочка не перегорит - 0.9998

> б) Вероятность наступления двух событий - И

In [21]:
p_l = 0.0004 * 0.0004
p_l

1.6e-07

In [22]:
Bp = puasson(2, p_l)
Bp

1.2799997952000166e-14

> вероятность что сгорят 2 лампы - 1.28 10-14

3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [35]:
Bp = bi_rasp(144, 70, 0.5)
Bp

0.06281178035144776

> Вероятность выпадения орла - 0.063

4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

> a)

In [24]:
#1
p11 = 0.7
p12 = 6 / 9
#2
p21 = 9 / 11
p22 = 8 / 10

# должны наступить два события
p_all = p11 * p12 * p21 * p22
p_all

0.3054545454545454

> вероятность что все белые = 0,305

б) Какова вероятность того, что ровно два мяча белые?

In [25]:
# два меча из 1, 0 из 2

a1_2 = 7 / 10 * 6 / 9
a2_0 = 2 / 11 * 1 / 10
pb1 = a1_2 * a2_0
pb1

0.008484848484848482

In [26]:
# 0 из 1, 2 из 2

a1_0 = 3 / 10 * 2 / 9
a2_2 = 9 / 11 * 8 / 10
pb2 = a1_0 * a2_2
pb2

0.04363636363636363

In [27]:
# по одному из каждой
pb3 = 7 / 10 * 9 / 11
pb3

0.5727272727272728

In [28]:
# общая сумма
p_ex_white = pb1 + pb2 + pb3
p_ex_white

0.6248484848484849

> в) В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча.  Какова вероятность того, что хотя бы один мяч белый?


> решаем от противного все - не белые
>
>

In [29]:
pb_1 = 3/10 * 2/9
pb_2 = 2/11 * 1/10
pb = pb_1 * pb_2
pb

0.0012121212121212121

In [30]:
1 - pb

0.9987878787878788

> вероятность что хотя бы 1 мяч будет белым = 0.9987